src/ProjectService.Business/Commands/Task/CreateTaskCommand.cs

```diff
@@ -16,93 +23,143 @@
+        _logger.LogWarning(
+          logMessage,
+          string.Join('\n', response.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage);
+      }
+
+      errors.Add("Can not create images. Please try again later.");
+
+      return null;
+    }
+
+    public CreateTaskCommand(
+      ITaskRepository repository,
+      ICreateTaskValidator validator,
+      IDbTaskMapper mapperTask,
+      IHttpContextAccessor httpContextAccessor,
+      IAccessValidator accessValidator,
+      IUserRepository userRepository,
+      IRequestClient<IGetCompanyEmployeesRequest> rcGetCompanyEmployees,
+      ILogger<CreateTaskCommand> logger,
+      IRequestClient<ICreateImagesRequest> rcImages)
+    {
+      _repository = repository;
+      _validator = validator;
+      _mapperTask = mapperTask;
+      _httpContextAccessor = httpContextAccessor;
+      _rcGetCompanyEmployee = rcGetCompanyEmployees;
+      _accessValidator = accessValidator;
+      _logger = logger;
+      _userRepository = userRepository;
+      _rcImages = rcImages;
+    }
 
-            Guid authorId = _httpContextAccessor.HttpContext.GetUserId();
+    public OperationResultResponse<Guid> Execute(CreateTaskRequest request)
+    {
+      OperationResultResponse<Guid> response = new();
 
-            if (!_accessValidator.IsAdmin()
-                && !_userRepository.AreUserProjectExist(authorId, request.ProjectId)
-                && GetDepartment(authorId, errors)?.DirectorUserId != authorId)
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
+      Guid authorId = _httpContextAccessor.HttpContext.GetUserId();
 
-            _validator.ValidateAndThrowCustom(request);
+      if (!_accessValidator.IsAdmin()
+        && !_userRepository.AreUserProjectExist(authorId, request.ProjectId)
+        && GetDepartment(authorId, response.Errors)?.DirectorUserId != authorId)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
 
-            Guid taskId = _repository.Create(_mapperTask.Map(request, authorId));
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Not enough rights.");
 
-            return new OperationResultResponse<Guid>
-            {
-                Body = taskId,
-                Status = !errors.Any() ?
-                    OperationResultStatusType.FullSuccess : OperationResultStatusType.PartialSuccess,
-                Errors = errors
-            };
-        }
+        return response;
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.AddRange(errors);
+
+        return response;
+      }
+
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
```

Comment:
you already create authorId with same data

Reply:
done